New Task topic: Update API Objects in Place Using kubectl patch

steveperry-53 opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/4519)
<!-- Reviewable:end -->

tengqm commented on docs/tasks/run-application/update-api-object-in-place.md:
```diff
@@ -0,0 +1,268 @@
+  imagePullPolicy: Always
+  name: patch-demo-ctr
+  ...
+  - image: redis
```
will line 88 cause indentation problem?

steveperry-53 replied:
Fixed.

tengqm commented on docs/tasks/run-application/update-api-object-in-place.md:
```diff
@@ -0,0 +1,268 @@
+want to update a list, you have to specify the entire new list. And the new list completely
+replaces the existing list.
+
+The `kubectl patch` command has a `type` parameter that you can set to to one of these values:
```
"set to to" -> "set to"

steveperry-53 replied:
Fixed

tengqm commented on docs/tasks/run-application/update-api-object-in-place.md:
```diff
@@ -0,0 +1,268 @@
+did a strategic merge patch.
+
+Next, do a JSON merge patch on your same Deployment. Create a file named patch-file-2.yaml
+that has this content:
```
patch-file-2.yaml -> `patch-file-2.yaml`

steveperry-53 replied:
Fixed

tengqm commented on docs/tasks/run-application/update-api-object-in-place.md:
```diff
@@ -0,0 +1,268 @@
+
+<table>
+  <tr><th>Parameter value</th><th>Merge type</th></tr>
+  <tr><td>json</td><td><a href="https://tools.ietf.org/html/rfc6902">JSON Patch, RFC 6902</a></td></tr>
```
Better add one sentence or two explaining how 'JSON' patch differs from 'JSON merge patch'.

steveperry-53 replied:
Added a link to a comparison article.

steveperry-53 commented on docs/concepts/cluster-administration/manage-deployment.md:
```diff
@@ -308,27 +308,6 @@ For more information, please see [kubectl scale](/docs/user-guide/kubectl/{{page
 
 Sometimes it's necessary to make narrow, non-disruptive updates to resources you've created.
 
-### kubectl apply
```
Fixed.

steveperry-53 commented on docs/tasks/run-application/update-api-object-in-place.md:
```diff
@@ -0,0 +1,268 @@
+---
+title: Update API Objects in Place Using kubectl patch
```
I want to keep this under Tasks, not Concepts. Yes, I think we should create other Update tasks and consolidate them under a heading. For now, let's leave it where it is.

I changed the filename to update-api-object-kubectl-patch.md

steveperry-53 replied:
Currently, [Managing Resources](https://kubernetes.io/docs/concepts/cluster-administration/manage-deployment/) is a mix of conceptual material and several tasks. I'l like to separate out the tasks into their own topics. Maybe under Tasks, we'll want to have a bucket for tasks about updating.

steveperry-53 commented:
@tengqm, Thanks for the review. I've addressed your comments in https://github.com/kubernetes/kubernetes.github.io/pull/4519/commits/3a4825e51c7d02b58fc7fb763f3ef25d649f5f51.
